thirdeye-ui/src/app/components/app-bar/app-bar-config-provider/app-bar-config-provider.component.tsx

```diff
@@ -26,21 +26,117 @@ import {
     AppBarConfigProviderProps,
     AppBarConfigProviderPropsContextProps,
 } from "./app-bar-config-provider.interface";
+import Appheader from "../app-header/app-header.component";
+import { useStyle } from "./app-bar.styles";
+import { useGetWorkspaces } from "../../../rest/workspace/workspace-action";
+import { useAuthV1 } from "../../../platform/stores/auth-v1/auth-v1.store";
+import { ActionStatus } from "../../../rest/actions.interfaces";
+import { notifyIfErrors } from "../../../utils/notifications/notifications.util";
+import { useTranslation } from "react-i18next";
+import { useNotificationProviderV1 } from "../../../platform/components";
 
 export const AppBarConfigProvider: FunctionComponent<AppBarConfigProviderProps> =
     ({ children }) => {
+        const { t } = useTranslation();
+        const { notify } = useNotificationProviderV1();
+        const { setWorkspace, workspace } = useAuthV1();
         const [showAppNavBar, setShowAppNavBar] = useState(true);
         const [okToRender, setOkToRender] = useState(false);
+        const [mainViewKey, setMainViewKey] = useState<string | null>(
+            "default"
+        );
+        const [mappedWorkspaces, setMappedWorkspaes] =
```

Comment:
why don't we separate this logic to worksapce switcher component

Reply:
I am trying to keep workspace switcher as dumb as possible. It should receive the options and display them without manipulating them.
Having said that, this is a temp thing because eventually there wont be any workspace with "null" as its id.